Library/Homebrew/test/testing_env.rb

```diff
@@ -20,6 +20,22 @@
   abort "Run `bundle install` or install the mocha and minitest gems before running the tests"
 end
 
+if ENV["HOMEBREW_TESTS_COVERAGE"]
+  # Ensure all Ruby files are `require`d for an accurate coverage measurement
+  library_base = File.expand_path("../..", __FILE__)
+  Dir["#{library_base}/**/*.rb"].each do |f|
+    f_rel = f[(library_base.length + 1)..-1] # path relative to `library_base`
+    # scripts
+    next if %w[build.rb postinstall.rb test.rb].include?(f_rel)
+    # mocked files
+    next if %w[config.rb global.rb].include?(f_rel)
+    # test files
+    next if f_rel.start_with?("test/")
```

Comment:
I think it would be a good idea to extend this to `next if f_rel.start_with?("compat/", "test/", "vendor/")` to match the filters we have in `.simplecov` and because we don't care about the coverage of those files. The comment above would require adjustment for this, too.

Additionally, we should probably exclude `compat.rb` to avoid pulling them in indirectly. (And because I think we don't want this compatibility stuff to affect testing.)